app/models/differential_expression_result.rb

```diff
@@ -0,0 +1,95 @@
+# store pointers to differential expression output sets for a given cluster/annotation
+class DifferentialExpressionResult
+  include Mongoid::Document
+  include Mongoid::Timestamps
+
+  # minimum number of observed_values, or cells per observed_value
+  MIN_OBSERVED_VALUES = 2
+
+  belongs_to :study
+  belongs_to :cluster_group
+
+  field :cluster_name, type: String # cache name of cluster at time of creation to avoid renaming issues
+  field :observed_values, type: Array, default: []
+  field :annotation_name, type: String
+  field :annotation_scope, type: String
+
+  validates :annotation_scope, inclusion: { in: %w[study cluster] }
+  validates :annotation_name, :cluster_name, presence: true
+  validate :has_observed_values?
+
+  before_validation :set_observed_values, :set_cluster_name
+
+  # pointer to source annotation object, either CellMetadatum of ClusterGroup#cell_annotation
+  def annotation_object
+    case annotation_scope
+    when 'study'
+      study.cell_metadata.by_name_and_type(annotation_name, 'group')
+    when 'cluster'
+      cluster_group.cell_annotations.detect do |annotation|
+        annotation[:name] == annotation_name && annotation[:type] == 'group'
+      end
+    end
+  end
+
+  # get query string formatted annotation identifier, e.g. cell_type__ontology_label--group--study
+  def annotation_identifier
+    case annotation_scope
+    when 'study'
+      annotation_object.annotation_select_value
+    when 'cluster'
+      cluster_group.annotation_select_value(annotation_object)
+    end
+  end
+
+  # compute the relative path inside a GCS bucket of a DE output file for a given label
+  def bucket_path_for(label)
+    "_scp_internal/differential_expression/#{filename_for(label)}"
+  end
+
+  # individual filename of label-specific result
+  def filename_for(label)
+    basename = [
+      cluster_name,
+      annotation_name,
+      label,
+      annotation_scope,
+      'wilcoxon'
```

Comment:
Is there a chance we'd want to change this type of DE - is there any case for having the analysis type as a param rather than hardcoded?

Reply:
Or maybe is there a case to store this value similar to `MIN_OBSERVED_VALUES` ?

Reply:
Or maybe is there a case to store this value similar to `MIN_OBSERVED_VALUES` ?